uk.co.compucorp.civicrm.hrleaveandabsences/sql/sample_data_cleanup.sql

```diff
@@ -0,0 +1,48 @@
+SET foreign_key_checks = 0;
+SELECT @staffID:=id FROM civicrm_contact WHERE display_name = 'civihr_staff@compucorp.co.uk';
+SELECT @managerID:=id FROM civicrm_contact WHERE display_name = 'civihr_manager@compucorp.co.uk';
+
+DELETE details,
+  job_leave,
+  health,
+  hour,
+  pay,
+  pension,
+  role,
+  revision,
+  contract
+FROM civicrm_hrjobcontract contract
+  LEFT JOIN civicrm_hrjobcontract_revision revision
+    ON contract.id = revision.jobcontract_id
+  LEFT JOIN civicrm_hrjobcontract_details details
+    ON details.jobcontract_revision_id = revision.id
+  LEFT JOIN civicrm_hrjobcontract_leave job_leave
+    ON job_leave.jobcontract_revision_id = revision.id
+  LEFT JOIN civicrm_hrjobcontract_health health
+    ON health.jobcontract_revision_id = revision.id
+  LEFT JOIN civicrm_hrjobcontract_hour hour
+    ON hour.jobcontract_revision_id = revision.id
+  LEFT JOIN civicrm_hrjobcontract_pay pay
+    ON pay.jobcontract_revision_id = revision.id
+  LEFT JOIN civicrm_hrjobcontract_pension pension
+    ON pension.jobcontract_revision_id = revision.id
+  LEFT JOIN civicrm_hrjobcontract_role role
+    ON role.jobcontract_revision_id = revision.id
+WHERE contract.contact_id = @staffID;
+
+TRUNCATE civicrm_hrleaveandabsences_absence_period;
+TRUNCATE civicrm_hrleaveandabsences_public_holiday;
+TRUNCATE civicrm_hrleaveandabsences_leave_period_entitlement;
+TRUNCATE civicrm_hrleaveandabsences_leave_request;
+TRUNCATE civicrm_hrleaveandabsences_leave_request_date;
+TRUNCATE civicrm_hrleaveandabsences_leave_balance_change;
```

Comment:
Thinking maybe we should delete only the data previously inserted into these tables by the sample data sql.

Joining leave_request to leave_request_date to balance_change table where the leave_request contact_id is @staffID can help get rid of leave request, leave request dates and their balance changes for the sample staff contact. Joining period_entitlement to balance changes too can help get rid of entitlements for the contact and the entitlement balance changes. For public holidays also, seems the ID's are sequential.

Reply:
I considered that, but I thought it was not worth. The main problem here are the Absence Periods. All the sample data depend on having the 2016 and the 2017 periods. That's why it creates those periods. So we have two options:
- Either delete the Absence Periods while clearing the sample data. In this case, there's no reason to keep all the other data, as there will be no period to contain them
- Or keep the Absence Period and delete the data. In this case, we have to consider two scenarios:
  - What happens if the user changes the absence period that we left in the DB?
  - We need to make the import script smarter to check if the period exists and only add them if they don't. To be honest, it's not only a matter of checking if they exist but also check if the dates are correct etc.

As you can see, it's a bit more complex than simply deleting the requests belonging to the staff. That's why I gave up and kept this working as it was before.

Note that this is different for contracts because the contract extension could have been installed before L&A and we don't want to delete data that was there previously

Reply:
Ok then. Inasmuch as sample data will not be installed on a real client site, I guess its fine then.